lib/event/emitter.js

```diff
@@ -39,7 +39,12 @@ EventEmitter.prototype.emit = function(type) {
   if (!listeners) {
     return;
   }
-  var args = Array.prototype.slice.call(arguments, 1);
+  // We need to copy arguments to local array for using it in another function call.
+  // The only way js engine (like V8) will be able to optimize it.
+  var args = [];
```

Comment:
instead of doing `args = []` it would be better to just allocate an array with the size of the arguments, and adding adding the arguments to the correct index.